cmd/traffic/cmd/manager/internal/state/intercept.go

```diff
@@ -0,0 +1,350 @@
+			wl.GetKind(), wl.GetName(), wl.GetNamespace())
+		if _, err := cmAPI.Update(ctx, cm, meta.UpdateOptions{}); err != nil {
+			return nil, fmt.Errorf("failed update entry for %s in ConfigMap %s.%s: %w", wl.GetName(), agent.ConfigMap, wl.GetNamespace(), err)
+		}
+		if ac, err = waitForConfigMapUpdate(ctx, cmAPI, wl.GetName(), wl.GetNamespace()); err != nil {
+			return nil, err
+		}
+	}
+	return ac, nil
+}
+
+func checkInterceptAnnotations(wl k8sapi.Workload) (bool, bool, error) {
+	pod := wl.GetPodTemplate()
+	a := pod.Annotations
+	if a == nil {
+		return false, true, nil
+	}
+
+	webhookEnabled := true
+	manuallyManaged := a[install.ManualInjectAnnotation] == "true"
+	ia := a[install.InjectAnnotation]
+	switch ia {
+	case "":
+		webhookEnabled = !manuallyManaged
+	case "enabled":
+		if manuallyManaged {
+			return false, false, errcat.User.Newf(
+				"annotation %s.%s/%s=enabled cannot be combined with %s=true",
+				wl.GetName(), wl.GetNamespace(), install.InjectAnnotation, install.ManualInjectAnnotation)
+		}
+	case "false", "disabled":
+		webhookEnabled = false
+	default:
+		return false, false, errcat.User.Newf(
+			"%s is not a valid value for the %s.%s/%s annotation",
+			ia, wl.GetName(), wl.GetNamespace(), install.ManualInjectAnnotation)
+	}
+
+	if !manuallyManaged {
+		return false, webhookEnabled, nil
+	}
+	cns := pod.Spec.Containers
+	var an *core.Container
+	for i := range cns {
+		cn := &cns[i]
+		if cn.Name == agent.ContainerName {
+			an = cn
+			break
+		}
+	}
+	if an == nil {
+		return false, false, errcat.User.Newf(
+			"annotation %s.%s/%s=true but pod has no traffic-agent container",
+			wl.GetName(), wl.GetNamespace(), install.ManualInjectAnnotation)
+	}
+	return false, true, nil
+}
+
+// Wait for the cluster's mutating webhook injector to do its magic. It will update the
+// configMap once it's done.
+func waitForConfigMapUpdate(ctx context.Context, cmAPI typed.ConfigMapInterface, agentName, namespace string) (*agent.Config, error) {
+	wi, err := cmAPI.Watch(ctx, meta.SingleObject(meta.ObjectMeta{
+		Name:      agent.ConfigMap,
+		Namespace: namespace,
+	}))
+	if err != nil {
+		return nil, fmt.Errorf("Watch of ConfigMap  %s failed: %w", agent.ConfigMap, ctx.Err())
+	}
+	defer wi.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			v := "canceled"
+			c := codes.Canceled
+			if ctx.Err() == context.DeadlineExceeded {
+				v = "timed out"
+				c = codes.DeadlineExceeded
+			}
+			return nil, status.Error(c, fmt.Sprintf("watch of ConfigMap %s[%s]: request %s", agent.ConfigMap, agentName, v))
+		case ev, ok := <-wi.ResultChan():
+			if !ok {
+				return nil, status.Error(codes.Canceled, fmt.Sprintf("watch of ConfigMap  %s[%s]: channel closed", agent.ConfigMap, agentName))
+			}
+			if !(ev.Type == watch.Added || ev.Type == watch.Modified) {
+				continue
+			}
+			if m, ok := ev.Object.(*core.ConfigMap); ok {
+				if y, ok := m.Data[agentName]; ok {
+					conf, ir := unmarshalConfigMapEntry(y, agentName, namespace)
```

Comment:
any reason this isn't called `err` as usual? scope? if so, can we call it something more clear like `confErr`?